### DISCUSSION TITLE
Multiple generators

### DISCUSSION THREAD

**Discussion by ben-xiao-io** [RobertCraigie/prisma-client-py#1029] [Q&A] [answered]
I have a monorepo with Python and Typescript. Is there a way to add both `prisma-client-js` and `prisma-client-py` generators in the same Prisma file?

I currently have this:

```
generator py {
    provider             = "prisma-client-py"
    recursive_type_depth = 5
}

generator client {
    provider = "prisma-client-js"
}

... rest of the schema
```

But running `prisma generate` results in this:

```
Prisma schema loaded from database\schema.prisma
Error: Could not resolve @prisma/client despite the installation that we just tried.
Please try to install it by hand with npm i @prisma/client and rerun npx "prisma generate" 🙏.
```

**Comment by ben-xiao-io** [comment, marked as answer]
If anyone else comes across this path, just add the flag `--generator py` to choose which generator to run

**Comment by felipeadeildo** [comment]
Hi, someone knows how to use different interface generators using the python-client-py?

I mean, something like that:

```typescript
generator client_async {
  provider               = "prisma-client-py"
  interface              = "asyncio"
  partial_type_generator = "models/partials_prisma.py"
  recursive_type_depth   = 5
}

generator client_sync {
  provider               = "prisma-client-py"
  interface              = "sync"
  partial_type_generator = "models/partials_prisma.py"
  recursive_type_depth   = 5
}
``` 

and generate both?

Cause I want to use the prisma client on celery tasks, but celery doens't have an official solution for async-functions tasks, so i can´t (in theory) use await expressions. But i want to use the async prisma client on FastAPI routes and on celery tasks use the sync prisma client...

How to do it? How to referenciate on a Prisma() instance which client i'm using?

**Reply by felipeadeildo** [reply]
Yo
Ive found a solution, i think i can just define a local output as

`prisma/client_async`
`prisma/client_sync`

and import one of them based on the context...
